phpunit/tests/fonts/font-library/wpFontCollection/getData.php

```diff
@@ -0,0 +1,277 @@
+		$this->assertSame(
+			$data->get_error_code(),
+			'font_collection_json_missing',
+			'Incorrect error code when invalid file path is provided.'
+		);
+	}
+
+	public function test_should_error_with_invalid_json_from_file() {
+		$mock_file = wp_tempnam( 'my-collection-data-' );
+		file_put_contents( $mock_file, 'invalid-json' );
+
+		$collection = new WP_Font_Collection( 'my-collection', $mock_file );
+
+		// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged -- Testing error response returned by `load_from_json`, not the underlying error from `wp_json_file_decode`.
+		$data = @$collection->get_data();
+
+		$this->assertWPError( $data, 'Error is not returned with invalid json file contents.' );
+		$this->assertSame(
+			$data->get_error_code(),
+			'font_collection_decode_error',
+			'Incorrect error code with invalid json file contents.'
+		);
+	}
+
+	public function test_should_error_with_invalid_url() {
+		$this->setExpectedIncorrectUsage( 'WP_Font_Collection::load_from_json' );
+
+		$collection = new WP_Font_Collection( 'my-collection', 'not-a-url' );
+		$data       = $collection->get_data();
+
+		$this->assertWPError( $data, 'Error is not returned when invalid url is provided.' );
+		$this->assertSame(
+			$data->get_error_code(),
+			'font_collection_json_missing',
+			'Incorrect error code when invalid url is provided.'
+		);
+	}
+
+	public function test_should_error_with_unsuccessful_response_status() {
+		add_filter( 'pre_http_request', array( $this, 'mock_request_unsuccessful_response' ), 10, 3 );
+
+		$collection = new WP_Font_Collection( 'my-collection', 'https://localhost/fonts/missing-collection.json' );
+		$data       = $collection->get_data();
+
+		remove_filter( 'pre_http_request', array( $this, 'mock_request_unsuccessful_response' ) );
+
+		$this->assertWPError( $data, 'Error is not returned when response is unsuccessful.' );
+		$this->assertSame(
+			$data->get_error_code(),
+			'font_collection_request_error',
+			'Incorrect error code when response is unsuccussful.'
+		);
+	}
+
+	public function test_should_error_with_invalid_json_from_url() {
+		add_filter( 'pre_http_request', array( $this, 'mock_request_invalid_json' ), 10, 3 );
+
+		$collection = new WP_Font_Collection( 'my-collection', 'https://localhost/fonts/invalid-collection.json' );
+		$data       = $collection->get_data();
+
+		remove_filter( 'pre_http_request', array( $this, 'mock_request_invalid_json' ) );
+
+		$this->assertWPError( $data, 'Error is not returned when response is invalid json.' );
+		$this->assertSame(
+			$data->get_error_code(),
+			'font_collection_decode_error',
+			'Incorrect error code when response is invalid json.'
+		);
+	}
+
+	public function mock_request( $preempt, $args, $url ) {
+		if ( 'https://localhost/fonts/mock-font-collection.json' !== $url ) {
+			return false;
+		}
+
+		return array(
+			'body'     => json_encode( self::$mock_collection_data ),
```

Comment:
does this need to be `wp_json_encode`?

Reply:
Good catch. Seems best to be consistent. Updated in https://github.com/WordPress/gutenberg/pull/58530/commits/6f0b8235bdd704668b7a3aa8e84c32e5fb31a5ad.